test/common/http/http1/codec_impl_test.cc

```diff
@@ -69,6 +115,76 @@ TEST_F(Http1ServerConnectionImplTest, Http10) {
 }
 
+TEST_F(Http1ServerConnectionImplTest, Http10AbsoluteNoOp) {
+  TestHeaderMapImpl expected_headers{{":path", "/"}, {":method", "GET"}};
+  Buffer::OwnedImpl buffer("GET / HTTP/1.0\r\n\r\n");
+  ExpectHeadersTest(Protocol::Http10, true, buffer, expected_headers);
+}
+
+TEST_F(Http1ServerConnectionImplTest, Http10Absolute) {
+  TestHeaderMapImpl expected_headers{
+      {":authority", "www.somewhere.com"}, {":path", "/foobar"}, {":method", "GET"}};
+  Buffer::OwnedImpl buffer("GET http://www.somewhere.com/foobar HTTP/1.0\r\n\r\n");
+  ExpectHeadersTest(Protocol::Http10, true, buffer, expected_headers);
+}
+
+TEST_F(Http1ServerConnectionImplTest, Http11AbsolutePath1) {
+  TestHeaderMapImpl expected_headers{
+      {":authority", "www.somewhere.com"}, {":path", "/"}, {":method", "GET"}};
+  Buffer::OwnedImpl buffer("GET http://www.somewhere.com/ HTTP/1.1\r\nHost: bah\r\n\r\n");
+  ExpectHeadersTest(Protocol::Http11, true, buffer, expected_headers);
+}
+
+TEST_F(Http1ServerConnectionImplTest, Http11AbsolutePath2) {
+  TestHeaderMapImpl expected_headers{
+      {":authority", "www.somewhere.com"}, {":path", "/"}, {":method", "GET"}};
+  Buffer::OwnedImpl buffer("GET http://www.somewhere.com/ HTTP/1.1\r\nHost: bah\r\n\r\n");
+  ExpectHeadersTest(Protocol::Http11, true, buffer, expected_headers);
+}
+
+TEST_F(Http1ServerConnectionImplTest, Http11AbsoluteEnabledNoOp) {
+  TestHeaderMapImpl expected_headers{
+      {":authority", "bah"}, {":path", "/foo/bar"}, {":method", "GET"}};
+  Buffer::OwnedImpl buffer("GET /foo/bar HTTP/1.1\r\nHost: bah\r\n\r\n");
+  ExpectHeadersTest(Protocol::Http11, true, buffer, expected_headers);
+}
+
+TEST_F(Http1ServerConnectionImplTest, Http11InvalidRequest) {
+  Buffer::OwnedImpl buffer("GET www.somewhere.com HTTP/1.1\r\nHost: bah\r\n\r\n");
+  Expect400(Protocol::Http11, true, buffer);
+}
+
+TEST_F(Http1ServerConnectionImplTest, Http11AbsolutePathNoSlash) {
+  TestHeaderMapImpl expected_headers{
+      {":authority", "www.somewhere.com"}, {":path", "/"}, {":method", "GET"}};
+  Buffer::OwnedImpl buffer("GET http://www.somewhere.com HTTP/1.1\r\nHost: bah\r\n\r\n");
+  ExpectHeadersTest(Protocol::Http11, true, buffer, expected_headers);
+}
+
+TEST_F(Http1ServerConnectionImplTest, Http11AbsolutePathBad) {
+  Buffer::OwnedImpl buffer("GET * HTTP/1.1\r\nHost: bah\r\n\r\n");
+  Expect400(Protocol::Http11, true, buffer);
+}
+
+TEST_F(Http1ServerConnectionImplTest, Http11AbsolutePortTooLarge) {
+  Buffer::OwnedImpl buffer("GET http://foobar.com:1000000 HTTP/1.1\r\nHost: bah\r\n\r\n");
+  Expect400(Protocol::Http11, true, buffer);
+}
+
+TEST_F(Http1ServerConnectionImplTest, Http11RelativeOnly) {
+  TestHeaderMapImpl expected_headers{
+      {":authority", "bah"}, {":path", "http://www.somewhere.com/"}, {":method", "GET"}};
+  Buffer::OwnedImpl buffer("GET http://www.somewhere.com/ HTTP/1.1\r\nHost: bah\r\n\r\n");
+  ExpectHeadersTest(Protocol::Http11, false, buffer, expected_headers);
+}
```

Comment:
I'm torn, I'd argue that its a nice switch to help prevent becoming an open proxy, but on the flipside there's nothing to prevent someone from abusing the reverse support. If thats not compelling I'll remove it.